File: LAB_03/src/ki304/salahub/Lab3/Driver.java
package ki304.salahub.Lab3;

import java.io.FileNotFoundException;
/**
 * The {@code Driver} class contains the main method for testing the functionality
 * of the {@link CopyMachine} class and its interactions with the {@link Scaner},
 * {@link Properties}, {@link Dimensions}, and {@link Functionality} classes.
 *
 * <p>It creates an instance of the {@code CopyMachine} class, performs various
 * operations such as changing the cartridge, turning on the power, obtaining
 * information about the scanner's dimensions, scanning photos and documents, and
 * printing them.</p>
 *
 * <p>Example usage:</p>
 * <pre>
 * {@code
 *   // Create an instance of CopyMachine
 *   CopyMachine cm = new CopyMachine();
 *
 *   // Test various operations
 *   cm.changeCardrige();
 *   cm.turnOnPower();
 *   cm.getInfoScanerDimensions();
 *   cm.changeCardrige();
 *   cm.scanPhotoAndSaveData("cat");
 *   cm.printPhoto("cat");
 *   cm.scanPhotoAndSaveData("cat");
 *   cm.printPhoto("cat");
 *   cm.scanDocumentAndSaveData("CV.docx");
 *   cm.printDoc("CV.docx");
 * }
 * </pre>
 *
 * @see CopyMachine
 * @see Scaner
 * @see Properties
 * @see Dimensions
 * @see Functionality
 * @author Andriy
 * @version 1.0
 * @since 2023-11-16
 */
public class Driver {
	/**
     * The main method for testing the functionality of the {@code CopyMachine} class.
     *
     * @param args The command-line arguments (not used in this context).
     * @throws FileNotFoundException if an error occurs while creating the log file.
     */
	public static void main(String[] args) throws FileNotFoundException {
		CopyMachine cm = new CopyMachine();
		cm.changeCardrige();
		cm.turnOnPower();
		cm.getInfoScanerDimensions();
		cm.changeCardrige();
		cm.scanPhotoAndSaveData("cat");
		cm.printPhoto("cat");
		cm.scanPhotoAndSaveData("cat");
		cm.printPhoto("cat");
		cm.scanDocumentAndSaveData("CV.docx");
		cm.printDoc("CV.docx");

	}

}

File: LAB_09/LAB_09/Driver.py
# Driver.py

# Importing the CopyMachine class
from CopyMachine import CopyMachine

# Main function to test the functionality of the CopyMachine class
def main():
    try:
        # Creating an instance of the CopyMachine class
        cm = CopyMachine()

        # Testing the change_cartridge method
        cm.change_cartridge()

        # Turning on the power of the CopyMachine
        cm.turn_on_power()

        # Retrieving and logging information about the scanner's dimensions
        cm.get_info_scaner_dimensions()

        # Attempting to change the cartridge while the CopyMachine is powered on
        cm.change_cartridge()

        # Scanning and saving photo data
        cm.scan_photo_and_save_data("cat")

        # Attempting to scan the same photo again
        cm.scan_photo_and_save_data("cat")

        # Retrieving and logging information about the scanner's properties
        cm.get_info_properties()

        # Turning off the power of the CopyMachine
        cm.turn_off_power()

    finally:
        # Closing the log file
        if cm.file:
            cm.file.close()

# Ensuring that the main function is executed only if the script is run, not imported
if __name__ == "__main__":
    main()
